data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/PostAggregateAndProofs.java

```diff
@@ -70,18 +110,26 @@ public PostAggregateAndProofs(
         @OpenApiResponse(status = RES_INTERNAL_ERROR, description = "Beacon node internal error.")
       })
   @Override
-  public void handle(Context ctx) throws Exception {
-    try {
-      final SignedAggregateAndProof[] signedAggregateAndProofs =
-          parseRequestBody(ctx.body(), SignedAggregateAndProof[].class);
+  public void handle(@NotNull Context ctx) throws Exception {
+    adapt(ctx);
+  }
+
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final List<SignedAggregateAndProof> signedAggregateAndProofs = request.getRequestBody();
+    final SafeFuture<List<SubmitDataError>> future =
+        provider.sendAggregateAndProofs(signedAggregateAndProofs);
```

Comment:
so we're retrieving this list but then using ErrorList still? maybe partial change, not sure, just confused...

Reply:
we should talk about the dependency... not sure